lib/internal/modules/esm/translators.js

```diff
@@ -352,14 +342,8 @@ translators.set('json', async function jsonStrategy(url) {
 });
 
 // Strategy for loading a wasm module
-translators.set('wasm', async function(url) {
+translators.set('wasm', async function(url, source) {
   emitExperimentalWarning('Importing Web Assembly modules');
-  let { source } = await this._getSource(
-    url, { format: 'wasm' }, defaultGetSource);
-  assertBufferSource(source, false, 'getSource');
-  ({ source } = await this._transformSource(
-    source, { url, format: 'wasm' }, defaultTransformSource));
-  assertBufferSource(source, false, 'transformSource');
```

Comment:
We need to keep this `assertBufferSource` here for Wasm loading.

Reply:
Ahh, sorry, got click-happy / copy+pasta (#35524 [removed it](https://github.com/nodejs/node/pull/35524/files#diff-b1de5e9ce1e9b411001c63d11a5092dbba32e1586269a20dc11408375eb2f16cL335), so I assumed it was supposed to be removed)

Reply:
Just wondering if we should apply this assertion to all load hooks anyway? Seems it was used in JSON too.

Reply:
Yes? Sorry, I don't have the context to provide input. On the surface, they're all supposed to support buffers, so that _sounds_ kosher.

Reply:
Looking at commonjs's translator, it doesn't actually use `source`. Should it? It currently check's ESMLoader's cjsCache or falls back to `CJSModule._load()`.

Reply:
Since `assertBufferSource` takes an `allowString` argument which is format specific, I guess it makes sense to validate source per translator. Marking as resolved.